assets/js/components/wp-dashboard/WPDashboardPopularPagesGA4.js

```diff
@@ -0,0 +1,163 @@
+import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
+import PreviewTable from '../PreviewTable';
+import TableOverflowContainer from '../TableOverflowContainer';
+import ReportTable from '../ReportTable';
+import DetailsPermaLinks from '../DetailsPermaLinks';
+import { numFmt } from '../../util';
+const { useSelect, useInViewSelect } = Data;
+
+export default function WPDashboardPopularPagesGA4( props ) {
+	const { WidgetReportError } = props;
+
+	const isGatheringData = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).isGatheringData()
+	);
+
+	const reportArgs = useSelect(
+		WPDashboardPopularPagesGA4.selectReportArgs,
+		[]
+	);
+
+	const report = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getReport( reportArgs )
+	);
+
+	const titles = useInViewSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getPageTitles( report, reportArgs )
+	);
+
+	const error = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getErrorForSelector( 'getReport', [
+			reportArgs,
+		] )
+	);
+
+	const loading = useSelect( ( select ) => {
+		const reportLoaded = select(
+			MODULES_ANALYTICS_4
+		).hasFinishedResolution( 'getReport', [ reportArgs ] );
+
+		const hasLoadedPageTitles = undefined !== error || undefined !== titles;
+
+		return ! hasLoadedPageTitles || ! reportLoaded;
+	} );
+
+	if ( loading || isGatheringData === undefined ) {
+		return <PreviewTable rows={ 6 } />;
+	}
+
+	if ( error ) {
+		return <WidgetReportError moduleSlug="analytics-4" error={ error } />;
+	}
+
+	// data.rows is not guaranteed to be set so we need a fallback.
+	const rows = report?.rows?.length ? cloneDeep( report.rows ) : [];
+	// Combine the titles from the pageTitles with the rows from the metrics report.
+	rows.forEach( ( row ) => {
+		const url = row.dimensionValues[ 0 ].value;
+		row.dimensionValues.unshift( { value: titles[ url ] } ); // We always have an entry for titles[url].
+	} );
+
+	return (
+		/* TODO: decouple the styles from search-console class */
+		<div className="googlesitekit-search-console-widget">
+			<h3>
+				{ __( 'Top content over the last 28 days', 'google-site-kit' ) }
+			</h3>
+			<TableOverflowContainer>
+				<ReportTable
+					rows={ rows }
+					columns={ tableColumns }
+					limit={ 5 }
+					gatheringData={ isGatheringData }
+					zeroState={ ZeroDataMessage }
+				/>
+			</TableOverflowContainer>
+		</div>
+	);
+}
+
+// Expose function for selecting report args for use in Storybook/tests.
+WPDashboardPopularPagesGA4.selectReportArgs = ( select ) => {
+	const dateRangeDates = select( CORE_USER ).getDateRangeDates( {
+		compare: true,
+		offsetDays: DATE_RANGE_OFFSET,
+	} );
+
+	return {
+		...dateRangeDates,
+		dimensions: [ 'pagePath' ],
+		metrics: [
+			{
+				name: 'screenPageViews',
+			},
+		],
+		orderby: [
+			{
+				metric: {
+					metricName: 'screenPageViews',
+				},
+				desc: true,
+			},
+		],
+		limit: 5,
+	};
+};
```

Comment:
This is an unusual approach and something that is not compatible with other components in the `wp-dashboard` folder. Let's move this back to the `WPDashboardPopularPagesGA4` component itself and add these arguments to the `wpDashboardAnalytics4OptionSets` list in the `common-GA4.stories.js` file to make it consistent with other GA4 components in this module.

https://github.com/google/site-kit-wp/blob/edc86a6b9a0020bc040d6dd6f33814523c6767fc/assets/js/components/wp-dashboard/common-GA4.stories.js#L32-L72